### PR TITLE
Various fixes to make unittests pass for windows platform.

### PR DESCRIPTION
**Pull request by dtao-oss**
1. First we needed to fix the libxml config to get libxml to compile.
2. Then the next problem is reading the test files.
	First the getrunfile returns a different location than on linux so I have to patch that.
  	It had problems reading files with a long path. Apparently windows has by default a limit of 256+ padding char for the path length. You have to edit a register value and also use a "\\?\" prefix trick to make it work.  The error message simply said file not found instead of something accurate. However, when you use the "\\?\" trick to "pass the path to the filesystem"  the automatic / to \ substitution disappears so you have to make sure you have \.   It's time like this I wanna rip out Bill's microchip outta me. Our current code probably still doesn't work with long paths but at least the tests should be ok.
3. Next is that gtest only supports "simple RE" on windows.  We didn't really need the full RE so rewriting the regex is good enough. However, that didn't solve the problem. The regex is used on the assert_death tests which uses regex to match stderr.  It turns out mini_chromium's CHECK() doesn't spit out anything to stderr on windows. So we have to modify and call maldoca::initlogging() explicitly in the tests to make it work.
4. After all that we can see the real problem.  It looks like the icu library is missing some code points. I think we covered some of the code points ourselves but it's missing utf-8 which is a must-have.
5. There are other failed tests in vba extract but they could be the consequence of not finding utf-8 code points. We have to fix icu first then see what happens.

### CODE REVIEW

File: maldoca/base/get_runfiles_dir.cc [hunk 1]
```diff
@@ -56,7 +56,10 @@ std::string GetExecutablePath() {
   HMODULE hModule = GetModuleHandleW(NULL);
   WCHAR wc_file_path[MAX_PATH] = {0};
   GetModuleFileNameW(hModule, wc_file_path, MAX_PATH);
-  std::string file_path = base::WideToUTF8(wc_file_path);
+  // see
+  // https://docs.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation?tabs=cmd
```
Comment (b-maldoca) [inline maldoca/base/get_runfiles_dir.cc:60]: fine for me, however seems this solution also has some issues: https://docs.bazel.build/versions/main/skylark/windows_tips.html#paths

they recommend using "--output_user_root=D:\bzl" or so.

if u wanna us the long paths, plz also add it to the project readme's requirements section :)

File: maldoca/ole/dir_test.cc [hunk 2]
```diff
@@ -71,9 +69,11 @@ TEST(OLEDirectoryEntryTest, BasicTest) {
   EXPECT_TRUE(entry.IsInitialized());
 
   // Can't initialize twice.
-  ASSERT_DEATH(entry.Initialize("foo", DirectoryStorageType::Storage, 1, 2, 3,
+   ASSERT_DEATH(entry.Initialize("foo", DirectoryStorageType::Storage, 1, 2, 3,
```
Comment (b-maldoca) [inline maldoca/ole/dir_test.cc:72]: remove space

File: maldoca/ole/dir_test.cc [hunk 3]
```diff
@@ -71,9 +69,11 @@ TEST(OLEDirectoryEntryTest, BasicTest) {
...
                                 4, 5, 6, clsid, 8, 9, 10),
                "Check failed: !is_initialized_");
+  /*entry.Initialize("foo", DirectoryStorageType::Storage, 1, 2, 3, 4, 5, 6,
```
Comment (b-maldoca) [inline maldoca/ole/dir_test.cc:75]: any reason to keep this code/comment?

File: maldoca/ole/dir_test.cc [hunk 4]
```diff
@@ -345,3 +345,9 @@ TEST_F(OLEDirectoryEntryTestFull, ChildrenAccess) {
 }
 
 }  // namespace
+
+int main(int argc, char **argv) {
```
Comment (b-maldoca) [inline maldoca/ole/dir_test.cc:349]: y do we need the main? or was this just for your manual testing?

File: maldoca/ole/endian_reader_test.cc [hunk 5]
```diff
@@ -78,7 +78,7 @@ TEST(EndianReaderTest, ReadLittleEndian) {
   EXPECT_TRUE(
       LittleEndianReader::ConsumeUInt64(&piece_eight_bytes, &eight_bytes));
   EXPECT_EQ(eight_bytes,
-            ((0x0aL << 56) + (0x0bL << 48) + (0x0cL << 40) + (0x0dL << 32) +
+            ((0x0aLL << 56) + (0x0bLL << 48) + (0x0cLL << 40) + (0x0dLL << 32) +
```
Comment (b-maldoca) [inline maldoca/ole/endian_reader_test.cc:81]: do u mind elaborating on this change?

File: maldoca/ole/dir_test.cc [hunk 6]
```diff
@@ -345,3 +345,9 @@ TEST_F(OLEDirectoryEntryTestFull, ChildrenAccess) {
...
+  ::testing::InitGoogleTest(&argc, argv);
+  maldoca::InitLogging();
```
Comment (b-maldoca) [inline maldoca/ole/dir_test.cc:351]: is the "#ifdef MALDOCA_CHROME" required here?

File: maldoca/ole/vba_extract_test.cc [hunk 7]
```diff
@@ -68,7 +68,7 @@
 #include "gtest/gtest.h"
```
Comment (b-maldoca) [inline maldoca/ole/vba_extract_test.cc:68]: what was changed in "maldoca/ole/testdata/ole/c98661bcd5bd2e5df06d3432890e7a2e8d6a3edcb5f89f6aaa2e5c79d4619f3d"? (change above)

File: third_party/libxml2_config/config.h [hunk 8]
```diff
@@ -121,7 +121,9 @@
 #define HAVE_PRINTF 1
 
 /* Define if <pthread.h> is there */
+#if !defined(_WIN32)
```
Comment (b-maldoca) [inline third_party/libxml2_config/config.h:124]: still confused y this has an effect as it shouldn't be included for windows: https://github.com/google/maldoca/blob/master/bazel/libxml.BUILD#L84

Comment (b-maldoca) [reply third_party/libxml2_config/config.h:124]: if u wanna keep it, plz also update: https://github.com/google/maldoca/tree/master/third_party/libxml2_config#readme

File: maldoca/base/file.h [hunk 9]
```diff
@@ -29,6 +30,7 @@
 #include "absl/strings/str_join.h"
 #include "absl/strings/string_view.h"
 #include "google/protobuf/message.h"
+#include "maldoca/base/get_runfiles_dir.h"
```
Comment (b-maldoca) [inline maldoca/base/file.h:33]: ./maldoca/base/file.h:33:10: fatal error: 'maldoca/base/get_runfiles_dir.h' file not found
#include "maldoca/base/get_runfiles_dir.h"

https://source.cloud.google.com/results/invocations/e6dc516f-3a02-494a-b4e4-89014cde6699/targets/maldoca%2Fgcp_ubuntu%2Fcc_test%2Fpresubmit/log

File: maldoca/base/file.cc [hunk 10]
```diff
@@ -249,7 +249,7 @@ StatusOr<std::string> GetContents(absl::string_view path) {
 
 #ifndef MALDOCA_CHROME
 absl::Status SetContents(const std::string& path, absl::string_view contents) {
-  auto fc = FileCloser(path, "w");
+  auto fc = FileCloser(path, "wb");
```
[the remark this answers is not in the document]
Comment (dtao-oss) [reply maldoca/base/file.cc:252]: It makes no difference on linux but can on windows.  windows used to (dunno if it still does) translate \n to \r\n in w mode.

[the remark this answers is not in the document]
Comment (b-maldoca) [reply maldoca/base/file.cc:252]: kk, maybe add a comment?

File: maldoca/base/file_test.cc [hunk 11]
```diff
@@ -48,6 +48,9 @@ TEST(Match, FindFiles) {
   auto file_2 = JoinPath(tmp_dir, "findfiles_1/abcd_123/xyz/file-2.txt");
   MALDOCA_ASSERT_OK(SetContents(file_1, "dummy"));
   MALDOCA_ASSERT_OK(SetContents(file_2, "dummy"));
+  std::string contents;
+  MALDOCA_ASSERT_OK(GetContents(file_1, &contents));
```
[the remark this answers is not in the document]
Comment (dtao-oss) [reply maldoca/base/file_test.cc:52]: oh ok, well we should test if the written stuff is read back correctly.

File: maldoca/base/get_runfiles_dir.cc (hunk 1 above)
Comment (dtao-oss) [reply maldoca/base/get_runfiles_dir.cc:60]: This is just for testing so don't really care. This is more portable;  presumably chrome may have the same long path problem and they don't use bazel.

File: maldoca/ole/dir_test.cc (hunk 2 above)
Comment (dtao-oss) [reply maldoca/ole/dir_test.cc:72]: werid clang didn't fix for me..

File: maldoca/ole/dir_test.cc (hunk 3 above)
Comment (dtao-oss) [reply maldoca/ole/dir_test.cc:75]: removed

File: third_party/libxml2_config/config.h (hunk 8 above)
Comment (b-maldoca) [reply third_party/libxml2_config/config.h:124]: let's keep it as is for now

File: maldoca/base/get_runfiles_dir.cc (hunk 1 above)
Comment (b-maldoca) [reply maldoca/base/get_runfiles_dir.cc:60]: k, added a todo for me to mention it in the reamdme.

File: maldoca/ole/vba_extract_test.cc (hunk 7 above)
Comment (dtao-oss) [reply maldoca/ole/vba_extract_test.cc:68]: put it back... arrgh windows defender!!!

**Review by b-maldoca — APPROVED** [review]

**Pull request merged**